sdk/attestation/azure-security-attestation/src/test/java/com/azure/security/attestation/AttestationPolicyTests.java

```diff
@@ -0,0 +1,285 @@
+                        logger.logExceptionAsError(new RuntimeException(e.toString()));
+                    }
+                    return null;
+                }).doOnNext(responseClaims -> {
+                    assertTrue(responseClaims.getClaims().containsKey("x-ms-policy-result"));
+                    assertEquals("Updated", responseClaims.getClaims().get("x-ms-policy-result").toString());
+                }).flatMap(responseClaims -> {
+                    String resetToken = null;
+                    switch (clientType) {
+                        case AAD:
+                            resetToken = new PlainPolicyResetToken().serialize();
+                            break;
+                        case ISOLATED:
+                            try {
+                                resetToken = new SecuredPolicyResetToken(signer, signingCertificateBase64).serialize();
+                            } catch (JOSEException e) {
+                                logger.logExceptionAsError(new RuntimeException(e.toString()));
+                            }
+                            break;
+                        default:
+                            throw new RuntimeException("Cannot ever hit this - unknown client type: " + clientType.toString());
+                    }
+                    return client.reset(attestationType, resetToken)
+                            .flatMap(response -> {
+                                try {
+                                    return verifyAttestationToken(httpClient, clientUri, response.getToken());
+                                } catch (ParseException e) {
+                                    logger.logExceptionAsError(new RuntimeException(e.toString()));
+                                }
+                                return null;
+                            });
+                })
+                .doOnNext(responseClaims -> {
+                    assertTrue(responseClaims.getClaims().containsKey("x-ms-policy-result"));
+                    assertEquals("Removed", responseClaims.getClaims().get("x-ms-policy-result").toString());
+                }))
+                .assertNext(claimSet -> { })
+                .verifyComplete();
+        }
+    }
+
+    /**
+     * Verifies the basic response for a Get Attestation Policy API - this simply verifies that the server
+     * returns a valid JWT and that the JWT contains a base64url encoded attestation policy.
+     * @param client HTTP Client, used when retrieving signing certificates.
+     * @param clientUri Client base URI - used when retrieving client signing certificates.
+     * @param attestationType attestation type - policy results vary by attestation type.
+     * @param policyResponse response from the getPolicy API.
+     */
+    private void verifyBasicGetAttestationPolicyResponse(HttpClient client, String clientUri, AttestationType attestationType, PolicyResponse policyResponse) throws ParseException {
```

Comment:
There are a lot of checked exceptions. Our guidelines is to wrap the checked exception in a RuntimeException so users don't have to add "throws" to their methods or handle it in a try/catch block.

https://azure.github.io/azure-sdk/java_introduction.html#java-errors-unchecked-exceptions